subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -43,26 +43,26 @@ static int comp_data_status(struct bt_mesh_model *model,
 			    struct net_buf_simple *buf)
 {
 	struct comp_data *param;
+	uint8_t status;
 	size_t to_copy;
 
 	BT_DBG("net_idx 0x%04x app_idx 0x%04x src 0x%04x len %u: %s",
 	       ctx->net_idx, ctx->app_idx, ctx->addr, buf->len,
 	       bt_hex(buf->data, buf->len));
 
-	if (!bt_mesh_msg_ack_ctx_match(&cli->ack_ctx, OP_DEV_COMP_DATA_STATUS, ctx->addr,
-				       (void **)&param)) {
-		return -ENOENT;
-	}
-
-	if (param->page) {
-		*(param->page) = net_buf_simple_pull_u8(buf);
-	}
+	status = net_buf_simple_pull_u8(buf);
```

Comment:
By pulling the status/page (u8) here and then pull an additional u8 if there is a page parameter the Composition Status will always be malformed.

Could simply rework to 
```
		if (param->page) {
			*(param->page) = net_buf_simple_pull_u8(buf);

		}else{
			status = net_buf_simple_pull_u8(buf);
		}
```

Reply:
Good point, I think I messed it up when I was resolving conflicts. I will post a patch asap.

Reply:
@michal-narajowski 
Don't worry that's why people like me run changes through real world tests just to double check, community effort.

Billy..